subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformer.java

```diff
@@ -195,5 +193,29 @@ public Object get(Type serviceType, Class<? extends Annotation> annotatedWith) t
             }
             return result;
         }
+
+        private static class InjectionPoint {
+            private final Class<? extends Annotation> annotation;
+            private final Class<?> injectedType;
+            private final Object valueToInject;
+
+            public InjectionPoint(Class<? extends Annotation> annotation, Class<?> injectedType, Object valueToInject) {
```

Comment:
We could bind `injectedType` to  the type of `valueToInject` to capture that these are about the same thing.